client/app/bundles/course/survey/containers/Respond.jsx

```diff
@@ -0,0 +1,185 @@
+        selected: isMultipleChoice ? (question_option_id.toString() === selected_option) : selected,
+      });
+      return ({ id, text_response, options_attributes: options.map(reduceOption) });
+    };
+    return {
+      response: {
+        answers_attributes: data.answers.map(reduceAnswer),
+        submit: data.submit,
+      },
+    };
+  }
+
+  static renderDescription(survey) {
+    if (!survey.description) {
+      return null;
+    }
+
+    return (
+      <Card>
+        <CardText>{survey.description}</CardText>
+      </Card>
+    );
+  }
+
+  componentDidMount() {
+    const { dispatch, params: { courseId, surveyId } } = this.props;
+    dispatch(fetchOwnResponse(courseId, surveyId));
+  }
+
+
+  handleUpdateResponse = (data) => {
+    const {
+      dispatch,
+      params: { courseId, surveyId },
+      responseForm: { responseId },
+    } = this.props;
+    const { saveSuccess, saveFailure, submitSuccess, submitFailure } = translations;
+    const payload = Respond.formatSurveyResponseData(data);
+    const successMessage = <FormattedMessage {...(data.submit ? submitSuccess : saveSuccess)} />;
+    const failureMessage = <FormattedMessage {...(data.submit ? submitFailure : saveFailure)} />;
+
+    return dispatch(
+      updateResponse(courseId, surveyId, responseId, payload, successMessage, failureMessage)
+    );
+  }
+
+  renderForm(response, survey, isLoading, pristine) {
+    if (isLoading) {
+      return <Subheader><FormattedMessage {...translations.loading} /></Subheader>;
+    }
+
+    return (
+      <div>
+        <Subheader><FormattedMessage {...surveyTranslations.questions} /></Subheader>
+        <ResponseForm
+          initialValues={Respond.buildInitialValues(response)}
+          onSubmit={this.handleUpdateResponse}
+          {...{ response, survey, pristine }}
+        />
+      </div>
+    );
+  }
+
+  render() {
+    const {
+      surveys,
+      responses,
+      responseForm: { isLoading, responseId },
+      params: { courseId, surveyId },
+      pristine,
+    } = this.props;
+
+    const findById = (array, id) => {
+      if (array && array.length > 0) {
+        return array.find(item => item.id.toString() === id.toString());
+      }
+      return {};
+    };
+    const survey = findById(surveys, surveyId);
+    const response = findById(responses, responseId);
+
+    return (
+      <div>
+        <TitleBar
+          title={survey.title}
+          iconElementLeft={<IconButton><ArrowBack /></IconButton>}
+          onLeftIconButtonTouchTap={() => browserHistory.push(`/courses/${courseId}/surveys`)}
+        />
+        {Respond.renderDescription(survey)}
+        {this.renderForm(response, survey, isLoading, pristine)}
+      </div>
+    );
+  }
+}
+
+Respond.propTypes = {
+  dispatch: PropTypes.func.isRequired,
+  params: PropTypes.shape({
+    courseId: PropTypes.string.isRequired,
+    surveyId: PropTypes.string.isRequired,
+  }).isRequired,
+  surveys: PropTypes.arrayOf(surveyShape),
```

Comment:
I'll move the selecting of the `survey` and `response` to `mapStateToProps`. Didn't realize before that `mapStateToProps` accepts `props` as its second parameter, so I did the selection in render because I needed access to the route params.